### DISCUSSION TITLE
Main console loop started..

### DISCUSSION THREAD

**Discussion by intera91** [OE4T/meta-tegra#1459] [Q&A]
A build that was working before no longer boots correctly 

I get a prompt  like this ] and keyboard is not responding in serial console

there is also this message main console loop started… 

I have no idea why or what the malfunction is

any hint on what is happening more than welcome
Sorry should be more precise

platform is Xavier NX on customised board. 
meta-tegra is dunfell 32.6.1

this build was  flashing and working fine but since I built another yocto with 35.4.1 which is working well then flashing this old build no longer boots correctly 

Does the 35.4.1 change something that affects 32.6.1 if flashed afterwards??? And how to correct that problem???

Thanks

**Comment by intera91** [comment]
According to an article in the nvidia forum it is possible that flashing 35.4.1 based meta-tegra yocto could have somehow corrupted or changed the content of the eprom and that would explain that after installing a UEFI based system the board can no longer use cboot based system

Could anyone confirm that and how to restore the eprom???

Thanks

**Reply by dwalkes** [reply]
> According to an article in the nvidia forum

Can you please share a link to the forum post?

I'm not aware of this, however note that https://github.com/OE4T/tegra-eeprom-tool is available for reading/writing the EEPROM.

**Reply by intera91** [reply]
https://forums.developer.nvidia.com/t/jetson-xavier-nx-won-t-load-gui-can-t-get-it-out-of-bash-environment/193551/35

**Reply by madisongh** [reply]
I don't see in that thread where it mentions EEPROM corruption being caused by 35.4.1, or anything coming out of a Yocto build.  If the CVM EEPROM has become corrupted, dumping its contents and comparing it to a working board might help in figuring out where the corruption is coming from.  

I've switched between R32 and R35-based builds on multiple Xavier dev kits with no problem. Last time I saw a question like this, it turned out the user had a program or driver that they had introduced themselves (for their custom board) that was overwriting part of the EEPROM.

As far as restoring the EEPROM, you can follow the info in that forum thread.  And as Dan mentions, we have a tool that can simplify the process.

**Reply by intera91** [reply]
I agree with the fact that this particular error isn’t mentioned… however I see similarities.

i think it is possible that the Eprom got corrupted by any i2c driver given that the underlying board is completely customised… 

I think it’s worth giving a shot to that theory as this is the only reasonable one I could come up with.

am still open to suggestions though…
